x11-misc/xscreensaver/xscreensaver-6.05.ebuild

```diff
@@ -0,0 +1,248 @@
+	x11-libs/libXext
+	x11-libs/libXi
+	x11-libs/libXrandr
+	x11-libs/libXt
+	x11-libs/libXxf86vm
+	elogind? ( sys-auth/elogind )
+	gdk-pixbuf? (
+		x11-libs/gdk-pixbuf-xlib
+		>=x11-libs/gdk-pixbuf-2.42.0:2
+	)
+	gtk? ( >=x11-libs/gtk+-2.22.0:3 )
+	jpeg? ( media-libs/libjpeg-turbo:= )
+	locking? ( virtual/libcrypt:= )
+	new-login? (
+		gdm? ( gnome-base/gdm )
+		!gdm? ( || ( x11-misc/lightdm lxde-base/lxdm ) )
+	)
+	opengl? (
+		virtual/glu
+		virtual/opengl
+	)
+	pam? ( sys-libs/pam )
+	png? ( media-libs/libpng:= )
+	systemd? ( >=sys-apps/systemd-221 )
+	xft? ( x11-libs/libXft )
+	xinerama? ( x11-libs/libXinerama )
+"
+# For USE="perl" see output of `qlist xscreensaver | grep bin | xargs grep '::'`
+RDEPEND="
+	${COMMON_DEPEND}
+	media-gfx/fbida
+	perl? (
+		dev-lang/perl
+		dev-perl/libwww-perl
+		virtual/perl-Digest-MD5
+	)
+	selinux? ( sec-policy/selinux-xscreensaver )
+"
+DEPEND="
+	${COMMON_DEPEND}
+	x11-base/xorg-proto
+"
+BDEPEND="
+	dev-util/intltool
+	sys-devel/bc
+	sys-devel/gettext
+	virtual/pkgconfig
+"
+PATCHES=(
+	"${FILESDIR}"/${PN}-6.01-interix.patch
+	"${FILESDIR}"/${PN}-5.31-pragma.patch
+	"${FILESDIR}"/${PN}-6.01-gentoo.patch
+	"${FILESDIR}"/${PN}-5.45-gcc.patch
+	"${FILESDIR}"/${PN}-6.01-configure.ac-sandbox.patch
+	"${FILESDIR}"/${PN}-6.01-without-gl-makefile.patch
+	"${FILESDIR}"/${PN}-6.01-non-gtk-install.patch
+	"${FILESDIR}"/${PN}-6.01-configure-install_sh.patch
+	"${FILESDIR}"/${PN}-6.03-without-gl-configure.patch
+	"${FILESDIR}"/${PN}-6.05-remove-update-icon-cache.patch
+)
+
+DOCS=( README{,.hacking} )
+
+src_prepare() {
+	default
+
+	sed -i configure.ac -e '/^ALL_LINGUAS=/d' || die
+	strip-linguas -i po/
+	export ALL_LINGUAS="${LINGUAS}"
+
+	if use new-login && ! use gdm; then #392967
+		sed -i \
+			-e "/default_l.*1/s:gdmflexiserver -ls:${EPREFIX}/usr/libexec/lightdm/&:" \
+			configure{,.ac} || die
+	fi
+
+	# We are patching driver/XScreenSaver.ad.in, so let's delete the
+	# header generated from it so that it gets back in sync during build:
+	rm driver/XScreenSaver_ad.h || die
+
+	if ! use offensive; then
+		sed -i \
+			-e '/boobies/d;/boobs/d;/cock/d;/pussy/d;/viagra/d;/vibrator/d' \
+			hacks/barcode.c || die
+		sed -i \
+			-e 's|erect penis|shuffle board|g' \
+			-e 's|flaccid penis|flaccid anchor|g' \
+			-e 's|vagina|engagement ring|g' \
+			-e 's|Penis|Shuttle|g' \
+			hacks/glx/glsnake.c || die
+		sed -i \
+			's| Stay.*fucking mask\.$||' \
+			hacks/glx/covid19.man \
+			hacks/config/covid19.xml || die
+		eapply "${FILESDIR}/xscreensaver-6.05-teach-handsy-some-manners.patch"
+	fi
+
+	config_rpath_update "${S}"/config.rpath
+
+	# Must be eauto*re*conf, to force the rebuild
+	eautoreconf
+}
+
+src_configure() {
+	if use ppc || use ppc64; then
+		filter-flags -maltivec -mabi=altivec
+		append-flags -U__VEC__
+	fi
+
+	unset BC_ENV_ARGS #24568
+
+	# Works similarly to -Werror,
+	# https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wimplicit-function-declaration
+	filter-flags -pedantic-errors
```

Comment:
Are people seriously putting this in *FLAGS?

I think we can safely drop this.. and if there's implicit function decl. errors, we want to fix those.

Reply:
Yeah dunno why you would put this in your flags. I dropped it, it's fine there are. 

This whole ebuild is full of wonders, it really is :D